### ISSUE TITLE
Refactor `NAICSContrastiveModel.training_step` for clarity

### ISSUE THREAD

**Issue by lowmason** [lowmason/naics-embedder#65]
## Summary

The `training_step` method in `NAICSContrastiveModel` spans hundreds of lines with many conditional branches handling various curriculum features. While logically correct, the method would benefit from decomposition into smaller, focused helper functions.

## Current State

The method handles all of the following in a single long sequence:
- Batch unpacking and forward pass
- Curriculum phase logic
- False negative masking
- Hard negative mining
- Negative sample statistics logging
- Loss computation
- Metric updates

This makes the method difficult to navigate, debug, and modify.

## Suggested Refactoring

Extract logical chunks into private helper methods:
```python
def training_step(self, batch, batch_idx):
    embeddings = self._forward_pass(batch)
    
    curriculum_config = self._get_curriculum_config()
    
    negatives, mask = self._prepare_negatives(
        embeddings,
        curriculum_config,
    )
    
    if self.hparams.use_false_negative_masking:
        mask = self._apply_false_negative_masking(batch, mask)
    
    if self._should_mine_hard_negatives(curriculum_config):
        negatives = self._perform_hard_negative_mining(
            embeddings,
            negatives,
            curriculum_config,
        )
    
    loss = self._compute_contrastive_loss(embeddings, negatives, mask)
    
    self._log_negative_sample_stats(negatives, mask)
    self._update_metrics(loss, embeddings)
    
    return loss
```

### Specific Extractions

| Current inline code | Suggested helper |
|---------------------|------------------|
| False negative mask computation | `_apply_false_negative_masking()` |
| Negative statistics gathering | `_log_negative_sample_stats()` |
| Hard mining branch | `_perform_hard_negative_mining()` |
| Curriculum phase checks | `_get_curriculum_config()` |
| Mining trigger logic | `_should_mine_hard_negatives()` |

## Benefits

- Each helper has a single responsibility
- Easier to unit test individual components
- Curriculum modifications don't require reading hundreds of lines
- Debugging can focus on specific helpers
- Consistent with existing patterns (`_compute_contrastive_loss`, utilities in `hard_negative_mining.py`)

## Priority

Low—this is a maintainability improvement, not a bug fix.
